### ISSUE TITLE
Cubes phasing into walls

### ISSUE THREAD

**Issue by CAD97** [Podshot/TwoTogether#4]
This one's an interesting one.

As is currently handled, the cubes have side colliders to determine when they can't move any further left/right so that they don't clip into walls.

However, when jumping against a wall, just that happens. The side collider thinks that it's no longer collided with the wall, allowing the cube to try to move into the wall, resulting in the buggy graphics, jumping back and forth.

It is easy to see the cause of this behavior by putting a Debug.Log in the CubeController script to see which sides are collided. The result is that, as hypothesized, jumping against a wall makes the side collider no longer collided with the wall.

As a side note: if rigidbody-based movement is used instead of transform-based movement, this leads to the cube being able to stick against the wall, rather than glitching into it graphically.

**Comment by CAD97** [comment]
Hopefully, a complete rewrite of movement logic won't be necessary. The use of side colliders to prevent this is a sound concept, it's just Unity's being weird with how the colliders are reporting their collision.

**Comment by CAD97** [comment]
Switching from OnEnter to OnStay seems to have removed this problem. I've also tweaked the positions of the colliders to be a bit cleaner, but that's maintenance rather than completely pertinent. Switching back to OnEnter leads to the same interaction.

It must be a Unity bug, probably with ordering; the colliders are still overlapping, however, OnExit was called more recently than OnEnter. Probably what's happening is sub-frame movement causes them both to be called in the same Unity internal update, in the order OnExit->OnEnter, but the order they're run on the colliders' script is OnEnter->OnExit.

Additionally, I have changed the side colliders to Trigger Colliders, as we don't really care about physics with their collisions.

After a bit more testing and tweaking, I'll commit this change. I also wonder if this will fix the physics movement as well...
